torchinductor/ir.py

```diff
@@ -1783,9 +1783,6 @@ def simplify_and_reorder(self):
 
         # the reordering_reindex in reads' simplify_reorder_and_tile
         reordering_reindex = [same_reorder(range(len(index_vars)))] * len(memory_addrs)
-        for i, reads_buf in enumerate(reads_bufs):
-            if isinstance(reads_buf, ComputedBuffer):
-                reordering_reindex[i] = reads_buf.iter_reordering_reindex
```

Comment:
@pyjhzwh I was hitting the error:
```
  File "/home/jansel/torchdynamo/torchinductor/scheduler.py", line 846, in __init__
    self.nodes.append(SchedulerNode(self, node, group_fn))
  File "/home/jansel/torchdynamo/torchinductor/scheduler.py", line 254, in __init__
    ) = node.simplify_and_reorder()
  File "/home/jansel/torchdynamo/torchinductor/ir.py", line 1788, in simplify_and_reorder
    reordering_reindex[i] = reads_buf.iter_reordering_reindex
AttributeError: 'ComputedBuffer' object has no attribute 'iter_reordering_reindex'
```

Is this the right fix for that?

Reply:
#862 seems the same

Reply:
Add try.. except block would solve this, #862 "AttributeError" would be gone
```
                try: 
                    reordering_reindex[i] = reads_buf.iter_reordering_reindex
                except AttributeError:
                    pass
```

Reply:
The root cause is a use node is iterated before a def node, so should we topo-sort `self.nodes` and keep it in the sorted order? https://github.com/pytorch/torchdynamo/blob/734141f29cf3dcc32a1a30695518063b19a5e4d6/torchinductor/scheduler.py#L838

Reply:
Sort by node's name should be ok. https://github.com/pytorch/torchdynamo/blob/66da4325abe6af7c59841ba0c8889cef65e094df/torchinductor/scheduler.py#L114